automation_tools/__init__.py

```diff
@@ -1763,22 +1774,25 @@ def create_openstack_instance(instance_name, image_name, flavor_name, ssh_key):
         network=network.id
     )
     # Assigning floating ip to instance
+    timeup = time.time()+(int(timeout)*60)
     while True:
         try:
+            if time.time() > timeup:
+                print('The timeout for assigning the floating IP has reached!')
+                sys.exit(1)
```

Comment:
Does this need to be in the `try` block? If reasonably possible, only code that might raise a `novaclient.exceptions.BadRequest` should be in the `try` block.